platform/linux-generic/odp_ishm.c

```diff
@@ -909,22 +910,36 @@ static int do_unmap(void *start, uint64_t size, uint32_t flags,
  * Returns the index of the found block (if any) or -1 if none.
  * Mutex must be assured by the caller.
  */
-static int find_block_by_name(const char *name)
+static int find_block_by_name(const char *name, odp_bool_t use_param_name)
 {
 	int i;
 
 	if (name == NULL || name[0] == 0)
 		return -1;
 
 	for (i = 0; i < ISHM_MAX_NB_BLOCKS; i++) {
-		if ((ishm_tbl->block[i].len) &&
-		    (strcmp(name, ishm_tbl->block[i].name) == 0))
+		const char *block_name = use_param_name ?
+					 ishm_tbl->block[i].param_name :
+					 ishm_tbl->block[i].name;
+
+		if (ishm_tbl->block[i].len && strcmp(name, block_name) == 0)
 			return i;
 	}
 
 	return -1;
 }
 
+/*
+ * Make application given memory block name unique
+ */
+static void name_to_ishm_name(const char *name, char *block_name)
```

Comment:
To me the name of this function does not match what it is doing and it got me confused. Maybe it could be renamed to something like make_block_name_unique and/or change the semantics a bit (e.g. call this make_unique_name and have it always generate the name, getting rid of the strcpy at the call site).

Reply:
Refactored in V2. Also, added missing `=` to while loop condition.